keras_nlp/models/llama/llama_attention.py

```diff
@@ -17,35 +17,34 @@
 from keras_nlp.utils.keras_utils import clone_initializer
 
 
-class LlamaAttention(keras.layers.Layer):
-    """Grouped query attention for Llama models"""
+class CachedLlamaAttention(keras.layers.Layer):
+    """A cached grounded query attention layer with sliding window."""
 
     def __init__(
         self,
         num_query_heads,
         num_key_value_heads,
+        rope_max_wavelength=10000,
         rope_scaling_factor=1.0,
         kernel_initializer="glorot_uniform",
-        rope_max_wavelength=10000,
-        max_sequence_length=512,
+        dropout=0,
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.num_query_heads = num_query_heads
-        self.num_key_value_heads = num_key_value_heads
+        self._num_query_heads = num_query_heads
```

Comment:
Can we keep the attrs that correspond to config/init args public? That is the style we want actually, even though we are not totally consistent here or in core Keras.

Reply:
Done.